### DISCUSSION TITLE
如何用python调用自己本地路径下的onnx模型

### DISCUSSION THREAD

**Discussion by yumianhuli2** [SWHL/RapidVideOCR#55] [Q&A]
如何用python调用自己本地路径下的onnx模型？，3Q！

**Comment by SWHL** [comment]
这个需要使用rapidocr_onnxruntime库，并且自己的模型要与PaddleOCR中 文本检测和识别一样。
详情参见：https://rapidai.github.io/RapidOCRDocs/docs/blog/06_custom_different_model/

**Comment by yumianhuli2** [comment]
> 这个需要使用rapidocr_onnxruntime库，并且自己的模型要与PaddleOCR中 文本检测和识别一样。 详情参见：https://rapidai.github.io/RapidOCRDocs/docs/blog/06_custom_different_model/
你好，示例里的字典地址是只支持英文的，有中文的或者更全的字典地址吗？并且我需要将这个字典地址放到哪里？

**Reply by SWHL** [reply]
rapidocr_onnxruntime默认就是中文的识别模型。字典文件已经写到onnx模型里面了。
文档写得有点模糊，已经更新说明了。 [docs](https://rapidai.github.io/RapidOCRDocs/docs/blog/06_custom_different_model/)

**Comment by yumianhuli2** [comment]
> rapidocr_onnxruntime默认就是中文的识别模型。字典文件已经写到onnx模型里面了。 文档写得有点模糊，已经更新说明了。 [docs](https://rapidai.github.io/RapidOCRDocs/docs/blog/06_custom_different_model/)

我用https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.7/doc/doc_ch/models_list.md里的ch_PP-OCRv4_server_det和ch_PP-OCRv4_server_rec高精度模型经过Paddle2ONNX命令成功转换后，替换rapidocr_onnxruntime库models文件夹下的原检测和推理模型并使用相同的命名，但是运行原来Work的代码报如下错哦：
![image](https://github.com/SWHL/RapidVideOCR/assets/39210837/d819a13b-56f5-4938-9ada-fffa3c75ccc7)

模型转换用到了如下的命令，但不清楚怎么将字典文件写到onnx模型里面（或许这就是报错的原因）？
paddle2onnx --model_dir ./inference/ch_PP-OCRv3_det_infer \
--model_filename inference.pdmodel \
--params_filename inference.pdiparams \
--save_file ./inference/det_onnx/model.onnx \
--opset_version 10 \
--enable_onnx_checker True

**Comment by SWHL** [comment]
要用我们写的那个转换工具，才能将字典写到里面去。你这里用的`paddle2onnx`是没有将字典写入到onnx中的

详情参见：https://rapidai.github.io/RapidOCRDocs/docs/blog/06_custom_different_model/

**Reply by yumianhuli2** [reply]
好的,3Q！还有比如我需要识别多语言，就是把多个字典拼接到一个字典里，然后再用转换工具写到模型里，是这个意思吗？

**Reply by SWHL** [reply]
如果要识别多种语言，这个就比较麻烦了。需要用到多种语言的文本检测模型，同时也需要用到不同语言的文本识别模型。
举个例子：如果你想识别英语、韩语字幕。需要什么：
1. 能够同时检测英语和韩语，并能区分是哪个字幕是英语、哪个是法语的文本检测模型
2. 需要单独的英语文本识别模型
3. 需要单独的韩语识别模型

**Reply by yumianhuli2** [reply]
好的，了解，3Q
